### PR TITLE
Shorten Prelude.Linear module documentation

### PR DESCRIPTION
**Pull request by mboes**

### CODE REVIEW

File: src/Prelude/Linear.hs [hunk 1]
```diff
@@ -1,16 +1,13 @@
 {-# LANGUAGE LinearTypes #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 
--- |
--- This module defines all the facilities you need when writing linear code. As
--- an analog to the original 'Prelude', it has the basic tools any haskell
--- program using @-XLinearHaskell@ will probably need.
---
--- This module is designed to be imported unqualifed.
+-- | This module redefines 'Prelude' data types, functions and type classes
```
Comment (aspiwack) [inline src/Prelude/Linear.hs:4]: It doesn't “redfine data types”, so the sentence, as it stands, is a bit misleading. Doc is hard, I'll come back to it soon, and attempt a fix.

**Comment by Divesh-Otwani** [comment]
Hope this works nicely ^

**Review by utdemir — APPROVED** [review]
LGTM. @aspiwack if you suggest an improvement over this, we can always send a separate PR.

**Comment by aspiwack** [comment]
Just, we don't provide linear versions of the data types: they are the same data types (though we add a handful of new ones which only make sense in linear contexts).

**Pull request merged**